app/org/sagebionetworks/bridge/play/controllers/UploadController.java

```diff
@@ -135,4 +145,28 @@ public Result uploadComplete(String uploadId, String synchronous) throws Excepti
         // Upload validation status may contain the health data record. Use the filter to filter out health code.
         return okResult(HealthDataRecord.PUBLIC_RECORD_WRITER, validationStatus);
     }
+    
+    public Result getUpload(String uploadId) throws Exception {
+        UserSession session = getAuthenticatedSession(Roles.RESEARCHER, Roles.ADMIN);
+
+        if (uploadId.startsWith("recordId:")) {
+            String recordId = uploadId.split(":")[1];
+
+            // This service does not throw an exception if the record is not found
+            HealthDataRecord record = healthDataService.getRecordById(recordId);
+            if (record == null) {
+                throw new EntityNotFoundException(HealthDataRecord.class);
+            }
+            uploadId = record.getUploadId();
+        }
+        UploadView uploadView = uploadService.getUploadView(uploadId);
+
+        // Even if the ID is valid, ultimately you cannot view an upload outside of your study
+        String studyId = session.getStudyIdentifier().getIdentifier();
+        if (!uploadView.getUpload().getStudyId().equals(studyId)) {
+            throw new UnauthorizedException();
+        }
```

Comment:
I removed this constraint. There may be certain things down the road that don't work in the Bridge Study Manager, like linking the health code back to a participant view, because the BSM doesn't really can't assume study switching like this. However the only pagination of uploads available for the UI is per study, so if you are browsing and not looking up a specific record regardless of the study you are in, it will all work as expected.